src/build/env.py

```diff
@@ -70,7 +70,7 @@ def _minimum_pip_version() -> str:
     return '19.1.0'
 
 
-def _has_valid_pip(purelib: str) -> bool:
+def _has_valid_pip(**distargs: object) -> bool:
```

Comment:
I'd rather we didn't lose the typing here, but Python doesn't make it easy:

```py
if typing.TYPE_CHECKING:
    from typing_extensions import NotRequired, TypedDict, Unpack

    class _Distargs(TypedDict):
        path: NotRequired[list[str]]

...

def _has_valid_pip(**distargs: Unpack[_Distargs]) -> bool:
```

Reply:
I left it along for now, would prefer a `_compat/typing` module if we add it.

Reply:
I can do that in a separate PR. I also usually have `_compat/tomllib` (https://github.com/scikit-build/scikit-build-core/tree/main/src/scikit_build_core/_compat for example).